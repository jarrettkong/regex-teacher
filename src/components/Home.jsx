import React from 'react'
import { NavLink } from 'react-router-dom';
import './Home.scss';

const Home = () => {
  return (
    <header className="Home-header">
      <h1 className="Home-name">/ Regex /</h1>
      <p className="Home-tagline">Learn Regular Expressions</p>
      <nav className="Home-btn-container">
        <NavLink to="/learn" className="App-btn"><a href="#Lesson" role="button">Learn</a></NavLink>
        <NavLink to="/sandbox" className="App-btn">Sandbox</NavLink>
        <p>Coming Soon</p>
      </nav>
    </header>
  )
}

export default Home;