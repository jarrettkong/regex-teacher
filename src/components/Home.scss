$accent-color: #00C296;
$background: #394950;

.Home {
  background-color: $background;
}

.Home-header {
  height: 100vh;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Home-name {
  font-size: 100px;
  margin: 0;
}

.Home-tagline {
  margin: 0 0 25px 0;
}

.Home-btn-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 25px;
  text-align: center;

  & p {
    grid-column-start: 2;
    text-transform: uppercase;
    margin-top: 10px;
    color: lightslategrey;
    font-weight: 600;
  }
}