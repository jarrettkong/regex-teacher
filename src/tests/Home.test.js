import React from 'react';
import Home from '../components/Home.jsx';
import { shallow } from 'enzyme';

describe('Home', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <Home />
    )
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
})