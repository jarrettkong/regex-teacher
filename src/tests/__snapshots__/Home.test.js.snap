// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<header
  className="Home-header"
>
  <h1
    className="Home-name"
  >
    / Regex /
  </h1>
  <p
    className="Home-tagline"
  >
    Learn Regular Expressions
  </p>
  <nav
    className="Home-btn-container"
  >
    <NavLink
      className="App-btn"
      to="/learn"
    >
      Learn
    </NavLink>
    <NavLink
      className="App-btn"
      to="/sandbox"
    >
      Sandbox
    </NavLink>
    <p>
      Work in Progress
    </p>
  </nav>
</header>
`;
